chain/fee_manager_test.go

```diff
@@ -0,0 +1,346 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	"time"
+	"testing"
+
+	"github.com/ava-labs/hypersdk/math"
+
+	"go.uber.org/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnitPrice(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Mock VM
+	rules := NewMockRules(ctrl)
+	rules.EXPECT().GetMinUnitPrice().Return(Dimensions{100, 100, 100, 100, 100})
+
+	// We set unit prices by the min unit price assigned at Genesis
+	// This is similar to a usage described in:
+	// https://github.com/ava-labs/hypersdk/blob/main/vm/vm.go#L326 
+	minUnitPrice := rules.GetMinUnitPrice()
+
+	var d Dimensions
+	feeManager := NewFeeManager(nil)
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		// Set unit prices for different dimensions 
+		feeManager.SetUnitPrice(i, minUnitPrice[i])
```

Comment:
Might want to make these all different and set a different value for last consumed.  I doubt it is the case that these are getting scrambled, but useful to confirm it.